python/cugraph/centrality/katz_centrality_wrapper.pyx

```diff
@@ -36,38 +34,28 @@ def katz_centrality(input_graph, alpha=0.1, max_iter=100, tol=1.0e-5, nstart=Non
     """
     Call katz_centrality
     """
-    cdef uintptr_t graph = graph_wrapper.allocate_cpp_graph()
-    cdef Graph * g = <Graph*> graph
+    if not input_graph.adjlist:
+        input_graph.view_adj_list()
 
-    if input_graph.adjlist:
-        [offsets, indices] = graph_wrapper.datatype_cast([input_graph.adjlist.offsets, input_graph.adjlist.indices], [np.int32])
-        [weights] = graph_wrapper.datatype_cast([input_graph.adjlist.weights], [np.float32, np.float64])
-        graph_wrapper.add_adj_list(graph, offsets, indices, weights)
-    else:
-        [src, dst] = graph_wrapper.datatype_cast([input_graph.edgelist.edgelist_df['src'], input_graph.edgelist.edgelist_df['dst']], [np.int32])
-        if input_graph.edgelist.weights:
-            [weights] = graph_wrapper.datatype_cast([input_graph.edgelist.edgelist_df['weights']], [np.float32, np.float64])
-            graph_wrapper.add_edge_list(graph, src, dst, weights)
-        else:
-            graph_wrapper.add_edge_list(graph, src, dst)
-        add_adj_list(g)
-        offsets, indices, values = graph_wrapper.get_adj_list(graph)
-        input_graph.adjlist = input_graph.AdjList(offsets, indices, values)
+    [offsets, indices] = graph_wrapper.datatype_cast([input_graph.adjlist.offsets, input_graph.adjlist.indices], [np.int32])
 
-    # we should add get_number_of_vertices() to Graph (and this should be
-    # used instead of g.adjList.offsets.size - 1)
-    num_verts = g.adjList.offsets.size - 1
+    num_verts = input_graph.number_of_vertices()
+    num_edges = len(indices)
```

Comment:
How about
```suggestion
    num_edges = input_graph.number_of_edges()
```

Reply:
input_graph.number_of_edges(), in the case of a Graph object (undirected Graph), returns len(indices)/2 (the number of unique unsymmetrized edges) based upon a python change that @Iroy30 put in during 0.13.  This num_edges variable needs to be the number of edges present in the COO/CSR data structure.

I originally had used the number_of_edges() function until Ishika's PR was merged.

Reply:
Makes sense. I forgot about this side effect.